c/CMakeLists.txt

```diff
@@ -25,30 +25,31 @@ add_library(kuliya SHARED kuliya.h)
 set_target_properties(kuliya PROPERTIES LINKER_LANGUAGE C)
 
 # Create the test executable
-add_executable(test test/test.c)
+add_executable(test test/test.cpp)
 
 # Create the example executable
 add_executable(example example/main.c)
 add_executable(example2 example/main.cpp)
 
 
-# Link the test and example executables with the "kuliya" library
-target_link_libraries(test kuliya)
+# Link the example executable with the "kuliya" library
 target_link_libraries(example kuliya)
 target_link_libraries(example2 kuliya)
 
 # Link the build and test executables with the "helpers" library
 target_link_libraries(build helpers)
-target_link_libraries(test helpers)
 
 # Find and link the jsmn library
 find_package(jsmn REQUIRED)
 target_link_libraries(build jsmn::jsmn)
 
 # Link the "helpers" shared library with libunistring
 find_package(libunistring REQUIRED)
-target_link_libraries(test libunistring::libunistring)
 target_link_libraries(build libunistring::libunistring)
 
+# Link test executable with required libraries
+find_package(CppUTest REQUIRED)
+target_link_libraries(test PRIVATE kuliya helpers cpputest::cpputest libunistring::libunistring)
+
```

Comment:
let's also properly layout the dependencies on run_build:
```suggestion

# Make all targets depend on the 'run_build' target, which in itself depends on the 'build' target
add_dependencies(kuliya run_build)
add_dependencies(test run_build)
add_dependencies(example run_build)
add_dependencies(example2 run_build)
```

otherwise if you run a fresh build (or delete `data.h`), you'll get this error:

```
build] /Users/zak/dev/dzcode-io/kuliya/c/./kuliya.h:4:10: fatal error: 'data.h' file not found
[build] #include "data.h"
[build]          ^~~~~~~~
[build] 1 error generated.
```